tests/test_fastapi_roles.py

```diff
@@ -0,0 +1,126 @@
+from typing import Awaitable, Callable, Tuple
+
+import pytest
+import sqlalchemy as sa
+from fastapi import FastAPI, HTTPException, Request, Response
+from fastapi.testclient import TestClient
+from starlette.middleware.base import BaseHTTPMiddleware
+
+# Public API
+from dbos import DBOS, DBOSContextEnsure
+
+# Private API because this is a unit test
+from dbos.context import assert_current_dbos_context
+from dbos.error import DBOSDuplicateWorkflowEventError, DBOSNotAuthorizedError
+
+
+def test_simple_endpoint(dbos_fastapi: Tuple[DBOS, FastAPI]) -> None:
+    dbos, app = dbos_fastapi
+
+    class SetRoleMiddleware(BaseHTTPMiddleware):
+        async def dispatch(
```

Comment:
Do we have to use an async function? Can we directly use the `@app.middleware()` decorator?